### ISSUE TITLE
tensor.max() does not behave equally in cuda and cpu in some cases

### ISSUE THREAD

**Issue by xvdp** [pytorch/pytorch#30708]
## 🐛 Bug

In some cases - as in the example below, tensor.max() fails to log the indices in the cuda result.

## To Reproduce

```python
import torch
rnd = torch.tensor([[ 0.0000, -1.0000, -1.0000, -1.0000, -1.0000, -1.0000, -1.0000, -1.0000,
         -1.0000, -1.0000, -1.0000],
        [ 0.5000, -1.0000, -1.0000, -1.0000, -1.0000, -1.0000, -1.0000, -1.0000,
         -1.0000, -1.0000, -1.0000],
        [ 8.0000,  8.0000,  8.0000,  8.0000,  8.0000,  8.0000,  8.0000, -1.0000,
         -1.0000, -1.0000, -1.0000],
        [ 7.0000,  7.0000,  7.0000,  7.0000,  7.0000,  7.0000, -1.0000, -1.0000,
         -1.0000, -1.0000, -1.0000]])

print(rnd.max(dim=1))
print((rnd.cuda()).max(dim=1))
""" my results: 
# CPU
torch.return_types.max(
values=tensor([0.0000, 0.5000, 8.0000, 7.0000]),
indices=tensor([0, 0, 6, 5]))
#CUDA
torch.return_types.max(
values=tensor([0.0000, 0.5000, 8.0000, 7.0000], device='cuda:0'),
indices=tensor([0, 0, 0, 0], device='cuda:0'))
"""
```

## Expected behavior

I'd've expected torch.max to return same order of max indices in  cuda and cpu.

## Environment

Is debug build: No
CUDA used to build PyTorch: 10.0.130

OS: Ubuntu 18.04.3 LTS
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
CMake version: version 3.10.2

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 10.1.243
GPU models and configuration: GPU 0: TITAN RTX
Nvidia driver version: 435.21
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.4

Versions of relevant libraries:
[pip] numpy==1.17.2
[pip] torch==1.3.0
[pip] torchaudio==0.2
[pip] torchfile==0.1.0
[pip] torchvision==0.4.1a0+d94043a
[conda] mkl                       2019.4                      243  
[conda] pytorch                   1.3.0           py3.6_cuda10.0.130_cudnn7.6.3_0    pytorch
[conda] torchaudio                0.2.0                    py36_1    pytorch
[conda] torchfile                 0.1.0                      py_0    conda-forge
[conda] torchvision               0.4.1                py36_cu100    pytorch

This is a special case, clearly, I encountered debugging some results. I would expect the indices to return the same preference: if one calls .max() on  a tensor, the index reutrned by cpu is the highest index of the highest value, in cuda, it is the lowest index of the highest value.

## Workaround found: argmax returns the same order of indices
```python
print("argmax")
print(torch.argmax(rnd, dim=1))

print("argmax cuda")
print(torch.argmax(rnd.cuda(), dim=1))

""" argmax results
argmax
tensor([0, 0, 6, 5])
argmax cuda
tensor([0, 0, 6, 5], device='cuda:0')
"""
```

cc @ezyang @gchanan @zou3519

**Comment by zou3519** [comment]
This is expected behavior. `torch.max` returns the index of the largest element; if there are multiple largest elements, it returns one of them. The behavior is different due to implementation differences on CPU and CUDA.

See https://github.com/pytorch/pytorch/issues/12779 for more details. 

I'm marking this as a docs issue for now: we should warn in the docs that it returns the index of one of the max elements but not necessarily the same one depending on the implementation.

**Comment by xvdp** [comment]
ok, admitedly i should be using different tool, in my case highest index of  highest value is what i was looking for, regardless of the repetition of high values.

but whats the point of indexing then if they will not return the same element? Id say either return same accross all platforms or remove indexing altogether.

btw. contrary to my comment,  argmax does not fix the issue. in returns highest index of first block of 8 elements. 
Nontheless, thankx for the clarification.

**Comment by albanD** [comment]
As discussed in triage review. Adding a warning to the doc is the way to go here.

**Comment by peterbell10** [comment]
@nikitaved, looks like your PR's [commit](https://github.com/pytorch/pytorch/commit/0414463007cecbc987ec376e95639073d26d7d2d) didn't auto-close this issue. `Fixes #30708` on its own should do the trick.
